leaderelection/context.go

```diff
@@ -168,9 +160,27 @@ func (b *standardBuilder) buildElector(ctx context.Context, la reconciler.Leader
 		// if lec.WatchDog != nil {
 		// 	lec.WatchDog.SetLeaderElection(le)
 		// }
-		buckets = append(buckets, &runUntilCancelled{Elector: le})
+		electors = append(electors, &runUntilCancelled{Elector: le})
+	}
+	return &runAll{les: electors}, nil
+}
+
+func newStandardBuckets(queueName string, cc ComponentConfig) []reconciler.Bucket {
+	names := make(sets.String, cc.Buckets)
+	for i := uint32(0); i < cc.Buckets; i++ {
+		names.Insert(standardBucketName(i, queueName, cc))
 	}
-	return &runAll{les: buckets}, nil
+	bs := hash.NewBucketSet(names)
+
+	bkts := make([]reconciler.Bucket, 0, cc.Buckets)
+	for name := range names {
+		bkts = append(bkts, hash.NewBucket(name, bs))
```

Comment:
I'm slightly confused here why the `hash.NewBucket` method isn't a method on the `hash.BucketSet`, but I don't think that needs to be fixed in this review.

Reply:
Didn't think of that API twist. Let me add it.

Reply:
🤣  it's what I originally suggested, but orthogonal to this.

With fresh eyes, I wonder whether it would make sense to also have a method that returns this entire list, which I think the autoscaler would also need for metrics forwarding to the appropriate shard, right?

Reply:
I think Yanwei builds that directly in the AS main,

Reply:
There's already a method `hash.BucketSet` returning the entire list.